dpc/src/parameters/testnet2.rs

```diff
@@ -77,6 +77,41 @@ macro_rules! dpc_setup {
     };
 }
 
+#[rustfmt::skip]
+macro_rules! dpc_snark_setup {
+    ($fn_name: ident, $static_name: ident, $snark_type: ident, $key_type: ident, $parameter: ident, $message: expr) => {
+        #[inline]
+        fn $fn_name() -> &'static <Self::$snark_type as SNARK>::$key_type {
+            static $static_name: OnceCell<<<Testnet2Parameters as Parameters>::$snark_type as SNARK>::$key_type> = OnceCell::new();
+            $static_name.get_or_init(|| {
+                <Self::$snark_type as SNARK>::$key_type::read_le(
+                    $parameter::load_bytes().expect(&format!("Failed to load parameter bytes for {}", $message)).as_slice()
+                ).expect(&format!("Failed to read {} from bytes", $message))
+            })
+        }
+    };
+}
+
+#[rustfmt::skip]
+macro_rules! dpc_snark_setup_with_mode {
+    ($fn_name: ident, $static_name: ident, $snark_type: ident, $key_type: ident, $parameter: ident, $message: expr) => {
+        #[inline]
+        fn $fn_name(is_prover: bool) -> &'static Option<<Self::$snark_type as SNARK>::$key_type> {
+            match is_prover {
+                true => {
+                    static $static_name: OnceCell<Option<<<Testnet2Parameters as Parameters>::$snark_type as SNARK>::$key_type>> = OnceCell::new();
+                    $static_name.get_or_init(|| {
+                        Some(<Self::$snark_type as SNARK>::$key_type::read_le(
+                            $parameter::load_bytes().expect(&format!("Failed to load parameter bytes for {}", $message)).as_slice(),
+                        ).expect(&format!("Failed to read {} from bytes", $message)))
+                    })
+                }
+                false => &None,
+            }
+        }
+    };
+}
+
```

Comment:
nit, can be done later: these macros can apply to both `testnet1` and `testnet2` with an extra added argument

Reply:
Discussed offline, attempted and unable to make it work for some odd reason. @ljedrz will investigate and we will follow up with a later PR if a resolution is found.